### ISSUE TITLE
Stage package not found in part 'packages-deb' (mysql-router / mysql-server-8.0)

### ISSUE THREAD

**Issue by Zvirovyi** [canonical/charmed-mysql-snap#57]
## Steps to reproduce
1. git clone
2. snapcraft pack

## Expected behavior
Snap packed successfully.

## Actual behavior
Encounter one of the next errors: `Stage package not found in part 'packages-deb': mysql-router=8.0.39-0ubuntu0.22.04.1.` or `Stage package not found in part 'packages-deb': mysql-server-8.0=8.0.39-0ubuntu0.22.04.1.`.

<img width="997" alt="image" src="https://github.com/user-attachments/assets/74b2f4e9-9015-4182-b62e-7ba447e353f8">

## Versions

Operating system: Ubuntu 24.04.1 LTS and Ubuntu 24.10

## Log output

[snapcraft-20241126-005801.161685.log](https://github.com/user-attachments/files/17909998/snapcraft-20241126-005801.161685.log)

## Additional context

Several days ago everything worked.

**Comment by paulomach** [comment]
@Zvirovyi thanks for the reported. Indeed odd issue. We will take a look

**Comment by Zvirovyi** [comment]
I've noticed, that snapcraft pulls the next version of these packages if I remove version constraints:
```
2024-12-03 03:54:50.619 :: 2024-12-03 01:50:49.356 Downloading package: mysql-server-8.0
2024-12-03 03:54:50.619 :: 2024-12-03 01:50:49.472 Get: 12 mysql-server-8.0_8.0.40-0ubuntu0.22.04.1_amd64.deb [1441 kB]
...
2024-12-03 03:54:50.620 :: 2024-12-03 01:50:54.518 Downloading package: mysql-router
2024-12-03 03:54:50.620 :: 2024-12-03 01:50:54.761 Get: 25 mysql-router_8.0.40-0ubuntu0.22.04.1_amd64.deb [4615 kB]
```
So, basically, repo has 8.0.40 version whereas snap restricted to the 8.0.39.
@paulomach

**Comment by paulomach** [comment]
Yes, just a matter of what is available on the repo at the time. I am closing it
